CONTRIBUTING.md

```diff
@@ -0,0 +1,207 @@
+## Contributing Guidelines
+
+Thank you for considering contributing to this project!
+To keep things smooth and maintain high quality, please follow the rules and best practices outlined below.
+
+- Be respectful and constructive in code reviews and discussions.
+- Follow the Code of Conduct at all times.
+- All code must be reviewed and approved via a Pull Request (PR)
+- Write clear, concise commit messages and PR descriptions.
+- Keep the ```main``` branch production-ready. Do not push directly to it.
+
+## Developer Workflow Overview
+
+- Assign an issue to a resource
+- Create a new branch to address the assigned issue
+- Address the issue
+- Commit changes
+- Open a pull request
+- Have a peer review the pull request
+    - A pull request may require revisions from the reviewer before being approved 
+- Merge approved pull request into the main branch
+- Delete the working branch if no longer needed
+
+## Coding Standards and Style Guides
+
+### General Principles (All Languages)
```

Comment:
Needs to include that first and foremost, match the style of the current context (file, package, etc)